src/main/java/net/coderbot/iris/uniforms/HardcodedCustomUniforms.java

```diff
@@ -50,6 +55,18 @@ private static SmoothedFloat rainStrengthS(FrameUpdateNotifier updateNotifier) {
 		return new SmoothedFloat(15, CommonUniforms::getRainStrength, updateNotifier);
 	}
 
+	private static float getRawPrecipitation() {
+		Biome.Precipitation precipitation = Minecraft.getInstance().level.getBiome(Minecraft.getInstance().player.blockPosition()).getPrecipitation();
+		switch (precipitation.getName()) {
+			case "rain":
+				return 1;
+			case "snow":
+				return 2;
+			default:
+				return 0;
+		}
```

Comment:
Please switch on the enum directly instead of converting it to a string first.

Reply:
Fixed, will push shortly after testing.